src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+import java.awt.*;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final double MAX_SIZE = 10.0;
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        String color = colorSupplier.getRandomColor();
+        switch (random.nextInt(5)) {
```

Comment:
Consider defining a constant for the number of figure types (5 in this case) to avoid using magic numbers. This will make the code more readable and easier to maintain. For example, you can define `private static final int FIGURE_TYPES_COUNT = 5;` and use it in `random.nextInt(FIGURE_TYPES_COUNT)`.